treetime/node_interpolator.py

```diff
@@ -68,7 +68,7 @@ def _convolution_integrand(t_val, f, g,
             tau = np.unique(np.concatenate((g.x, t_val-f.x,[tau_min,tau_max])))
         else:
             tau = np.unique(np.concatenate((g.x, f.x-t_val,[tau_min,tau_max])))
-        tau = tau[(tau>tau_min-ttconf.TINY_NUMBER)&(tau<tau_max+ttconf.TINY_NUMBER)]
+        np.clip(tau, tau_min-ttconf.TINY_NUMBER, tau_max+ttconf.TINY_NUMBER)
```

Comment:
did this change lead to a large speed-up? In the previous implementation, `tau` could shrink. Now we might end up with several identical values. what about pulling the `np.unique` out of the `if` statements above and wrapping `np.clip` in the `np.unique`?